src/cloud-api-adaptor/test/e2e/assessment_runner.go

```diff
@@ -545,6 +551,33 @@ func (tc *TestCase) Run() {
 				}
 
 			}
+
+			if tc.alternateImageName != "" {
+				var podlist v1.PodList
+				var podLogString string
+				expectedSuccessMessage := "Choosing " + tc.alternateImageName
+				if err := client.Resources("confidential-containers-system").List(ctx, &podlist); err != nil {
+					t.Fatal(err)
+				}
+				for _, pod := range podlist.Items {
+					if pod.Labels["app"] == "cloud-api-adaptor" {
+						podLogString, _ = GetPodLog(ctx, client, pod)
+						break
+					}
+				}
+				if strings.Contains(podLogString, expectedSuccessMessage) {
+					t.Logf("PodVM was brought up using the alternate PodVM image %s", tc.alternateImageName)
+				} else {
+					t.Logf("cloud-api-adaptor pod logs: %s", podLogString)
+					yamlData, err := yaml.Marshal(tc.pod.Status)
+					if err != nil {
+						log.Errorf("Error marshaling pod.Status to JSON: %s", err.Error())
+					} else {
+						t.Logf("Current Pod State: %v", string(yamlData))
+					}
+					t.Fatal("failed due to unknown reason")
+				}
```

Comment:
Optional request: Could we move this section to assessment_helper.go as a new function? I'm trying to reduce how massive the assessment_runner has got with this strategy in some WIP commits I'm working on on the side.